### PR TITLE
Add a policy linter.

### PR DESCRIPTION
**Pull request by briantkennedy**

### CODE REVIEW

**Comment by t12g** [comment]
Are the test failures expected?

**Comment by briantkennedy** [comment]
Nope, looks like the strictness found some issues in the test data, will fix.

**Comment by briantkennedy** [comment]
Bumping.

File: cmd/policy-tool/lint/lint.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package lint
+
+import (
+	"github.com/forseti-security/config-validator/pkg/gcv"
+	"github.com/spf13/cobra"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "lint",
+	Short:   "Lint a directory continaing ConstraintTemplates and/or Constraints.",
```
Comment (t12g) [inline cmd/policy-tool/lint/lint.go]: sp: containing

File: cmd/policy-tool/lint/lint.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+var (
+	flags struct {
+		policies []string
+		libs     string
+	}
+)
+
+func init() {
+	Cmd.Flags().StringSliceVar(&flags.policies, "policies", nil, "Path to the policies directory.")
```
Comment (t12g) [inline cmd/policy-tool/lint/lint.go]: this can be > 1 directory right? mention that in the description?

File: cmd/policy-tool/lint/lint.go [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"github.com/forseti-security/config-validator/pkg/gcv"
+	"github.com/spf13/cobra"
+)
+
+var Cmd = &cobra.Command{
+	Use:     "lint",
+	Short:   "Lint a directory continaing ConstraintTemplates and/or Constraints.",
+	Example: `policy-tool status --path ./forseti-security/policy-library/policies`,
```
Comment (t12g) [inline cmd/policy-tool/lint/lint.go]: update the example usage?

File: pkg/gcv/configs/config.go [hunk 4]
```diff
@@ -103,7 +104,7 @@ func ListRegoFiles(dir string) ([]string, error) {
 	return arrayFilterSuffix(files, ".rego"), nil
 }
 
-func loadUnstructured(dirs []string) ([]*unstructured.Unstructured, error) {
+func LoadUnstructured(dirs []string) ([]*unstructured.Unstructured, error) {
```
Comment (t12g) [inline pkg/gcv/configs/config.go]: can we add go doc comments for these exported funcs?

File: pkg/gcv/configs/config.go [hunk 5]
```diff
@@ -43,14 +44,14 @@ func init() {
 }
 
 const (
-	constraintGroup = "constraints.gatekeeper.sh"
+	ConstraintGroup = "constraints.gatekeeper.sh"
 	expectedTarget  = "validation.gcp.forsetisecurity.org"
-	yamlPath        = expectedTarget + "/yamlpath"
+	YamlPath        = expectedTarget + "/yamlpath"
```
Comment (t12g) [inline pkg/gcv/configs/config.go]: nit: dunno if we're following google style here... if we are, it should be YAMLPath per go/gocomments#initialisms

File: cmd/policy-tool/lint/lint.go (hunk 1 above)
Comment (briantkennedy) [reply cmd/policy-tool/lint/lint.go]: done

File: cmd/policy-tool/lint/lint.go (hunk 3 above)
Comment (briantkennedy) [reply cmd/policy-tool/lint/lint.go]: done

File: cmd/policy-tool/lint/lint.go (hunk 2 above)
Comment (briantkennedy) [reply cmd/policy-tool/lint/lint.go]: done

File: pkg/gcv/configs/config.go (hunk 5 above)
Comment (briantkennedy) [reply pkg/gcv/configs/config.go]: lowercased it.

File: pkg/gcv/configs/config.go (hunk 4 above)
Comment (briantkennedy) [reply pkg/gcv/configs/config.go]: done

**Pull request merged**